browser/net/brave_proxying_web_socket.cc

```diff
@@ -395,6 +396,7 @@ void BraveProxyingWebSocket::OnHeadersReceivedComplete(int error_code) {
   if (proxy_has_extra_headers()) {
     proxy_trusted_header_client_->OnHeadersReceived(
         headers,
+        ::net::IPEndPoint(),
```

Comment:
seems like we should be doing something else here? cc @iefremov

Reply:
If we wanted to, I believe we can get the actual endpoint object inside `BraveProxyingWebSocket::OnConnectionEstablished`, from the `response` object via something like this:

```diff
diff --git a/browser/net/brave_proxying_web_socket.cc b/browser/net/brave_proxying_web_socket.cc
index dd163a943..660e79bde 100644
--- a/browser/net/brave_proxying_web_socket.cc
+++ b/browser/net/brave_proxying_web_socket.cc
@@ -187,6 +187,7 @@ void BraveProxyingWebSocket::OnConnectionEstablished(
     mojo::ScopedDataPipeConsumerHandle readable) {
   DCHECK(forwarding_handshake_client_);
   DCHECK(!is_done_);
+  remote_endpoint_ = response->remote_endpoint;
   forwarding_handshake_client_->OnConnectionEstablished(
       std::move(websocket), std::move(client_receiver), selected_protocol,
       extensions, std::move(response), std::move(readable));
@@ -367,7 +368,7 @@ void BraveProxyingWebSocket::OnHeadersReceivedComplete(int error_code) {
   if (proxy_has_extra_headers()) {
     proxy_trusted_header_client_->OnHeadersReceived(
         headers,
-        ::net::IPEndPoint(),
+        remote_endpoint_,
         base::BindOnce(
             &BraveProxyingWebSocket::OnHeadersReceivedCompleteFromProxy,
             weak_factory_.GetWeakPtr()));
diff --git a/browser/net/brave_proxying_web_socket.h b/browser/net/brave_proxying_web_socket.h
index de688d98f..233180d9c 100644
--- a/browser/net/brave_proxying_web_socket.h
+++ b/browser/net/brave_proxying_web_socket.h
@@ -147,6 +147,7 @@ class BraveProxyingWebSocket : public network::mojom::WebSocketHandshakeClient,
   network::ResourceRequest request_;
   network::ResourceResponseHead response_;
   scoped_refptr<net::HttpResponseHeaders> override_headers_;
+  net::IPEndPoint remote_endpoint_;

   GURL redirect_url_;
   bool is_done_ = false;
```

Based off code in https://cs.chromium.org/chromium/src/extensions/browser/api/web_request/web_request_proxying_websocket.cc?q=WebRequestProxyingWebSocket&dr=CSs&l=176

Reply:
4545e2d2526b6e146e279760651c1e9e07997fff: added the change suggested by @bsclifton and also applied another code change from web_request_proxying_websocket.